### PR TITLE
[FEATURE] Add a privacy policy checkbox to subscription form

### PR DESCRIPTION
**Pull request by the-coding-owl**
This adds a checkbox to the subscription forms, so the user can
accept it.

### CODE REVIEW

File: Configuration/TypoScript/setup.txt [hunk 1]
```diff
@@ -216,6 +216,11 @@ plugin.tx_t3extblog {
 				# checks if client accepts cookies
 				cookie = 2
 			}
+
+            privacyPolicy {
+                enabled = {$plugin.tx_t3extblog.settings.privacyPolicy.enabled}
```
Comment (fnagel) [inline Configuration/TypoScript/setup.txt]: We probably want to have this more specific to enable / disable specific checkboxes. So probably more like `enableForBlogSubscription`?

File: Documentation/Configuration/Index.rst [hunk 2]
```diff
@@ -61,6 +61,8 @@ settings.latestComments                     Some settings for the latest comment
 settings.rss                                Basis rss-settings (see :code:`/Configuration/TypoScript/RSS/setup.txt` for page config)
 settings.backend                            Configure the backend output
 settings.debug                              Some debug configs
+settings.privacyPolicy.enabled              Enable the privacy policy checkbox in the subscription forms
```
Comment (fnagel) [inline Documentation/Configuration/Index.rst]: This says "forms" but it's actually only for the blog post subscription form, right?

**Comment by the-coding-owl** [comment]
Hi @fnagel 

Thank you for the kind words about my PR :)

I am no lawyer and this change is based on a customer of ours, who wanted to have this checkbox.
As one interprets the GDPR (https://gdpr-info.eu/art-4-gdpr/ see point 11), the visitor must give a clear sign of agreement with usage of his data, thus the safest would indeed be a checkbox to be clicked separately.
But I am not sure about that, because this regulation lacks clear examples and we need to wait for a court to decide the details.
I made this checkbox hideable, because I do not know whether it would be needed or not, so for people who do not want to have it, it can be disabled or vise versa.

As for the "subscribe to new comments" feature: Sure, I might just have forgotten that.

**Comment by fnagel** [comment]
This PR has been modified and merged in ab04ee1d1ebf4af2b2666e43cf45569b3a2b8991

Thanks for your contribution!

**Pull request closed without merging**